src/legacy/core_plugins/interpreter/server/routes/server_functions.js

```diff
@@ -90,15 +90,19 @@ function runServerFunctions(server) {
         try {
           const result = await runFunction(server, handlers, fnCall);
 
-          if (result != null) {
-            return { id, statusCode: 200, result };
+          if (typeof result === 'undefined') {
+            return batchError(id, `Function ${fnCall.functionName} did not return anything.`);
           }
 
-          return batchError(id, `Function ${fnCall.functionName} did not return anything.`);
+          return { id, statusCode: 200, result };
         } catch (err) {
           if (Boom.isBoom(err)) {
             return batchError(id, err.output.payload, err.statusCode);
+          } else if (err instanceof Error) {
```

Comment:
NOTE: this is the only error type I've run in to (never saw a Boom error), and I'm pretty sure we encourage users to throw from functions, so I don't think there are any other conditions to handle.